clippy_utils/src/paths.rs

```diff
@@ -161,3 +161,4 @@ pub const OPTION_UNWRAP: [&str; 4] = ["core", "option", "Option", "unwrap"];
 pub const OPTION_EXPECT: [&str; 4] = ["core", "option", "Option", "expect"];
 pub const FORMATTER: [&str; 3] = ["core", "fmt", "Formatter"];
 pub const DEBUG_STRUCT: [&str; 4] = ["core", "fmt", "builders", "DebugStruct"];
+pub const ORD_CMP: [&str; 4] = ["core", "cmp", "Ord", "cmp"];
```

Comment:
Wouldn't [`sym::cmp`](https://doc.rust-lang.org/beta/nightly-rustc/rustc_span/symbol/sym/constant.cmp.html) work?

Reply:
No, take a look at this code for example: https://play.rust-lang.org/?version=nightly&mode=debug&edition=2021&gist=a9afbef379712a643187be77ae228d12

If it both defines a method `cmp` and also implements `Ord`, and `Ord` is not in scope (due to `#![no_implicit_prelude]`), it'll call the method instead - *not* `Ord::cmp`. Yet it still doesn't lint

*But* the suggestion is wrong and should be fixed...